### PR TITLE
Add scoped slot for Subheader to Table Component

### PR DESCRIPTION
**Pull request by nickyg91**
Fixes #
This PR adds scoped slots for a subheader to the table component.
Addresses issue #92.

## Proposed Changes
Added subheader scoped slot to the table component for both oruga and oruga-next.
- Added element to contain the subheader
- Added function hasCustomSubheader to hide/show slot if there is/is not a subheader.

### CODE REVIEW

**Comment by nickyg91** [comment]
Below is a screenshot of the subheader in a non-mobile context.
image
Below is a screenshot of the mobile table - it keeps its original look and feel.
image

File: packages/oruga-next/src/components/table/Table.vue [hunk 1]
```diff
@@ -135,6 +135,39 @@
...
+                    <th v-if="showDetailRowIcon" width="40px" />
+                    <th v-if="checkable && checkboxPosition === 'left'" />
+                    <th
+                        v-for="(column, index) in visibleColumns"
+                        :key="column.newKey + ':' + index + 'subheading'"
+                        :style="column.style"
+                    >
+                        <div
+                            :class="{
+                                'is-numeric': column.numeric,
```
Comment (jtommy) [inline packages/oruga-next/src/components/table/Table.vue:148]: `numeric` is not relative to a specific style at the moment (you can remove it) and `centered` doesn't exist, we use a generic position prop that you can move on above `th` (you might convert the supposed `thSubheadingClasses` from computed to a method) and remove the `div`

**Review by jtommy — CHANGES_REQUESTED** [review]
@nickyg91 Thanks for the contribution 😉 
I've just reviewed it, probably you took inspiration from Buefy (and it's ok of course) but please keep in mind the Oruga concept that I described in the single comments.

File: packages/oruga-next/src/components/table/Table.vue [hunk 2]
```diff
@@ -135,6 +135,39 @@
                     </th>
                     <th v-if="checkable && checkboxPosition === 'right'" />
                 </tr>
+                <tr v-if="hasCustomSubheadings" class="o-table__th">
+                    <th v-if="showDetailRowIcon" width="40px" />
```
[the remark this answers is not in the document]
Comment (nickyg91) [reply packages/oruga-next/src/components/table/Table.vue:139]: I also noticed there was another width="40px"  further up. Would you like me to remove that as well?
image

[the remark this answers is not in the document]
Comment (jtommy) [reply packages/oruga-next/src/components/table/Table.vue:139]: Yes!

**Comment by nickyg91** [comment]
Subheadings after suggested changes implemented.
image

File: packages/oruga-next/src/components/table/Table.vue (hunk 1 above)
Comment (nickyg91) [reply packages/oruga-next/src/components/table/Table.vue]: @jtommy I believe i got this done correctly based off your feedback, but am still uncertain. Let me know if I need to make a change here after you view the new changes.

**Review by jtommy — APPROVED** [review]
@nickyg91 thanks, well done!

**Pull request merged**